crates/transaction-pool/src/validate/mod.rs

```diff
@@ -30,10 +30,12 @@ pub enum TransactionValidationOutcome<T: PoolTransaction> {
         state_nonce: u64,
         /// Validated transaction.
         transaction: T,
+        /// Whether to propagate the transaction to the network.
+        propagate: bool,
     },
     /// The transaction is considered invalid indefinitely: It violates constraints that prevent
     /// this transaction from ever becoming valid.
-    Invalid(T, InvalidPoolTransactionError),
+    Invalid(T, InvalidPoolTransactionError, bool),
```

Comment:
here we don't need it, because we never want to propagate an invalid tx in the first place